src/core/loader.cpp

```diff
@@ -0,0 +1,24 @@
+// SPDX-FileCopyrightText: Copyright 2024 shadPS4 Emulator Project
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include "common/io_file.h"
+#include "common/types.h"
+#include "loader.h"
+
+FileTypes DetectFileType(const std::string& filepath) {
+    if (filepath.size() == 0) // no file loaded
+    {
+        return FileTypes::Unknown;
+    }
+    Common::FS::IOFile file;
+    file.Open(filepath, Common::FS::FileAccessMode::Read);
+    file.Seek(0);
+    u32 magic;
+    file.ReadRaw<u8>(&magic, sizeof(magic));
+    file.Close();
+    switch (magic) {
+    case 0x544e437f: // PS4 PKG
```

Comment:
This should be a compile time constant. Maybe something like ```constexpr static u32 PkgMagic = 0x544e437f;```

Reply:
ok